src/routes.js

```diff
@@ -1,43 +1,75 @@
+// @flow
+
+import type RoutesSettingsStore, {
+  RouteSettings,
+} from './stores/RoutesSettingsStore';
+
 import * as React from 'react';
 import {
   DrawerNavigator,
   StackNavigator,
   TabNavigator,
 } from 'react-navigation';
+import { inject } from 'mobx-react';
 
 import MainDrawer from './components/MainDrawer';
 
 import LoginScreen from './screens/LoginScreen';
 import SplashScreen from './screens/SplashScreen';
 
 import HomeScreen from './screens/HomeScreen';
+import SettingsScreen from './screens/SettingsScreen';
 import MyLocationsScreen from './screens/MyLocationsScreen';
 
-// todo login should be own stacknavigator with login/register screens
+type Props = {|
+  routesSettingsStore: RoutesSettingsStore,
+|};
+
+@inject('routesSettingsStore')
 /* eslint-disable sorting/sort-object-props */
-const rootRouter = StackNavigator(
-  {
-    splash: { screen: SplashScreen },
-    login: { screen: LoginScreen },
-    main: {
-      screen: DrawerNavigator(
-        {
-          home: { screen: HomeScreen },
-          myLocations: { screen: MyLocationsScreen },
+export class AppRouter extends React.Component<Props> {
+  _addInitialRouteSettings = (
```

Comment:
I wish I could do something better, but this is  the only pattern which works for me. I hoped that I can set initial params to route, but I can't. I also tried to use HOC requireManageTap where I setup static navigationOption, for hiding item in drawer and decorate particular screens with it, but then it works as expected only after I visit that screen. Also, here I can't just recreate rootRouter and rerender AppRouter on settings changes, since it will reset all navigation history.

Reply:
I think you're overthinking how this should work. I think the routes should always be there but you would just have to hide buttons in the menu. That's how I am doing it in the ionic app.

Either way, this is fine for now. I'd rather have you working on features and pages instead of fighting with frameworks.

Reply:
Yeah, in my current solution, routes are always here, with this function I just add settings to the routes, like initialParams(idk why react-navigation still doesn't support that..there are big issies about it in their repo). and in drawer I just react on this params and hide some of the buttons.